service/r2/store/kv/store.go

```diff
@@ -366,28 +401,32 @@ func (s *store) DeleteRollupRule(
 func (s *store) FetchRollupRuleHistory(namespaceID, rollupRuleID string) ([]*models.RollupRuleView, error) {
 	rs, err := s.ruleStore.ReadRuleSet(namespaceID)
 	if err != nil {
-		return nil, s.handleUpstreamError(err)
+		return nil, handleUpstreamError(err)
 	}
 
 	rrs, err := rs.RollupRules()
 	if err != nil {
-		return nil, s.handleUpstreamError(err)
+		return nil, handleUpstreamError(err)
 	}
 
 	hist, exists := rrs[rollupRuleID]
 	if !exists {
-		return nil, s.rollupRuleNotFoundError(namespaceID, rollupRuleID)
+		return nil, rollupRuleNotFoundError(namespaceID, rollupRuleID)
 	}
 	return hist, nil
 }
 
 func (s *store) Close() { s.ruleStore.Close() }
 
-func (s *store) newUpdateMeta(uOpts r2.UpdateOptions) rules.UpdateMetadata {
+func (s *store) fetchRuleSet(namespaceID string) (rules.RuleSet, error) {
```

Comment:
Let's inline this method where it's used (only in `UpdateRuleSet` it seems) and remove it to reduce cognitive burden.